### ISSUE TITLE
Resolve certificate issue

### ISSUE THREAD

**Issue by jmmcnj** [18F/dol-whd-14c#335]
As an application user, I would like be able to use the application on IE 11 - fix error:
```
SCRIPT7002: XMLHttpRequest: Network Error 0x2ef3, Could not complete the operation due to error 00002ef3
```
See the links below:
https://stackoverflow.com/questions/23145688/ie10-11-ajax-xhr-error-script7002-xmlhttprequest-network-error-0x2ef3/32106649#32106649
https://stackoverflow.com/questions/16312938/ie10-ie11-abort-post-ajax-request-after-clearing-cache-with-error-network-error/16818527#16818527
http://jonnyreeves.co.uk/2013/making-xhr-request-to-https-domains-with-winjs

#### Functionality

- [ ] Obtain domain name or sufficient work around SCRIPT7002: XMLHttpRequest: Network Error

**Comment by klinden** [comment]
@mgwalker @mmurthydol @jmmcnj Given our difficulty with getting a valid SSL certificate in the Integration/QA environments is anybody opposed to running the application over HTTP in these environments? We have config settings for both the UI and API that can control this so no code change would be needed. From my end I don't see any issues this would cause.

**Comment by klinden** [comment]
See #331 for some context.

**Comment by mgwalker** [comment]
@klinden Sounds good to me.  Sorry for the slow response.

What does that mean for the future of #331?  Do we close it rather than make temporary code changes?

**Comment by klinden** [comment]
@mgwalker No worries. Yes, we can close #331 as it should not be needed.

@davoodharun Can you please setup the Integration and QA environments to use HTTP (both UI and API) and confirm that this issue has been resolved?

**Comment by klinden** [comment]
I've been reading up some more on this and while I think resolving the certificate issue was part of the problem I don't think it fully resolves this issue (and this is confirmed by the fact that Michelle is still having issues). I think @davoodharun was on the right track with #331 as it seems that if the first call that the browser makes is a POST that the SSL certificates are not loaded properly (even if they are valid). I'm wondering though if instead of baking this extra call into the authentication service logic if we could somehow put it in the app startup logic to just make a GET call to a "hello world" endpoint (for IE only). I know that Angular 2/4 has a concept of an App Initializer which I've used before and has worked pretty well. I'm not sure if that is something we can do in an app using ngUpgrade. Thoughts @mgwalker @davoodharun?

**Comment by mgwalker** [comment]
Huh.  What an oddball bug.  I haven't spent much time reading about it like you have, but just a couple of Google searches suggest you're on to it.  Several people have encountered issues with IE11's handling of CORS preflight requests in precisely this scenario (web app and API at different URLs, both served over SSL).  I'm not sure if Angular's `$http` does the preflight `OPTIONS` request (it looks like it does) but if so, IE's behavior there could be the culprit.  Great find!

If sending a load-time `GET` request to the API will resolve the issue, that seems reasonable to me.  Unfortunately, in 2017, we still have to work around implementation defects in 4-year-old browsers.  😬   (And since IE 11's support life is tied to Windows 10's, that means we get to keep working around it through at least 2025.  Yaaaaaaay!)

**Comment by davoodharun** [comment]
I believe @binwang89 said on today's call that she was able to log into the application using IE11 without getting that error. There was an underlying issue that was addressed with #355 that was preventing Michelle from logging in.
